khmer/utils.py

```diff
@@ -18,9 +18,89 @@ def print_error(msg):
     print >>sys.stderr, msg
 
 
+def check_is_pair(record1, record2):
+    """
+    Checks to see if the two sequence records are left and right pairs
+    of each other, respectively.  Returns True or False as appropriate.
+
+    Handles both Casava formats: seq/1 and seq/2, and 'seq::... 1::...'
+    and 'seq::... 2::...'.
+    """
+    is_fastq = False
+    if hasattr(record1, 'accuracy') or hasattr(record2, 'accuracy'):
+        if not (hasattr(record1, 'accuracy') and hasattr(record2, 'accuracy')):
+            raise ValueError("both records must be same type (FASTA or FASTQ)")
+        is_fastq = True
+
+    name1 = record1.name
+    name2 = record2.name
+
+    if is_fastq and ' ' in name1:                        # handle '@name 1:rst'
```

Comment:
On Tue, Feb 10, 2015 at 02:15:04PM -0800, Camille Scott wrote:
> > +    Checks to see if the two sequence records are left and right pairs
> > +    of each other, respectively.  Returns True or False as appropriate.
> > +
> > +    Handles both Casava formats: seq/1 and seq/2, and 'seq::... 1::...'
> > +    and 'seq::... 2::...'.
> > +    """
> > +    is_fastq = False
> > +    if hasattr(record1, 'accuracy') or hasattr(record2, 'accuracy'):
> > +        if not (hasattr(record1, 'accuracy') and hasattr(record2, 'accuracy')):
> > +            raise ValueError("both records must be same type (FASTA or FASTQ)")
> > +        is_fastq = True
> > +
> > +    name1 = record1.name
> > +    name2 = record2.name
> > +
> > +    if is_fastq and ' ' in name1:                        # handle '@name 1:rst'
> 
> Sure, but plenty of people strip the quality line from their sequence at
> some point in their pipeline

I do not believe that to be true; most mappers use it, and all assemblers will
take it even if they don't use it.

But, you know what? let me ask :)